ci: add workflow to sync with upstream daily

HastD opened the pull request:
This workflow runs the following steps daily:

* Pull tags from upstream;
* Rebase the default branch onto the latest tag for the current Fedora version;
* Push the changes to the default branch, including tags.

If the rebase encounters conflicts, the job will abort and conflicts will need to be resolved manually.

RoyalOughtness commented on .github/workflows/upstream-sync.yml:
```diff
@@ -0,0 +1,45 @@
+      contents: write # Needed to modify repository contents
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          persist-credentials: true
+          fetch-depth: 0
```
@HastD this can be ignored like so, if needed https://github.com/secureblue/secureblue/blob/e84b5fd81673f2aea2e231a86be789e0c31e58f1/.github/workflows/build-one-recipe.yml#L121C103-L121C162

HastD committed:
ci: add workflow to sync with upstream daily

This workflow runs the following steps daily:

* Pull tags from upstream;
* Rebase the default branch onto the latest tag for the current Fedora
  version;
* Push the changes to the default branch, including tags.

If the rebase encounters conflicts, the job will abort and conflicts
will need to be resolved manually.

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>
